_rules/link-alone-descriptive-aizyf1.md

```diff
@@ -0,0 +1,222 @@
+
+```html
+<span role="link" tabindex="0" onclick="document.location+='#desc'">See description of the product.</span>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 4
+
+This link has an [accessible name][] (from its `aria-labelledby` attribute) describing its purpose.
+
+```html
+<p id="instructions">Click on the arrow to go to the main content.</p>
+<svg xmlns="http://www.w3.org/2000/svg" x="0" y="0">
+	<a href="#main" aria-labelledby="instructions">
+		<path
+			style="fill:#1E201D;"
+			d="M21.205,5.007c-0.429-0.444-1.143-0.444-1.587,0c-0.429,0.429-0.429,1.143,0,1.571l8.047,8.047H1.111
+			C0.492,14.626,0,15.118,0,15.737c0,0.619,0.492,1.127,1.111,1.127h26.554l-8.047,8.032c-0.429,0.444-0.429,1.159,0,1.587
+			c0.444,0.444,1.159,0.444,1.587,0l9.952-9.952c0.444-0.429,0.444-1.143,0-1.571L21.205,5.007z"
+		/>
+	</a>
+</svg>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This link has an [accessible name][] which does not describe its purpose.
+
+```html
+<a href="#desc">More</a>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Failed Example 2
+
+This link has an [accessible name][] which does not describe its purpose.
+
+```html
+<div role="link" tabindex="0" onclick="document.location+='#main'">More</div>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Failed Example 3
+
+This link has an [accessible name][] which does not describe its purpose.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" x="0" y="0">
+	<a href="#main">
+		<text x="20" y="20">
+			Go
+		</text>
+	</a>
+</svg>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Failed Example 4
+
+This link has an [accessible name][] which, alone, does not describe its purpose.
+
+```html
+<p>See the description of <a href="#desc">this product</a>.</p>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 5
+
+These links have [accessible name][] which, alone, do not describe their purpose.
+
+```html
+<ul>
+	<li>
+		Ulysses
+		<ul>
+			<li><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm"> HTML </a></li>
+			<li>
+				<a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73">
+					EPUB
+				</a>
+			</li>
+			<li><a href="https://www.gutenberg.org/files/4300/4300-0.txt"> Plain text </a></li>
+		</ul>
+	</li>
+</ul>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no [semantic link][] in this document.
+
+```html
+<a href="https://www.w3.org/WAI" role="button">Web Accessibility Initiative (WAI)</a>
+```
+
+#### Inapplicable Example 2
+
+This link is not [included in the accessibility tree][].
+
+```html
+<a href="https://www.w3.org/WAI" style="display: none;"
+	><img src="/test-assets/5effbb/cart.svg" alt="Checkout" />Checkout</a
+>
+```
+
+#### Inapplicable Example 3
```

Comment:
Don't think we need this TBH. If you can inapplicable 1 and 2 right, you won't get this wrong.

Reply:
Hmm, ok, not a deal breaker. Feel free to leave it in.